core/string/ustring.cpp

```diff
@@ -3744,8 +3743,7 @@ int String::_count(const char *p_string, int p_from, int p_to, bool p_case_insen
 			return 0;
 		}
 		if (p_from == 0 && search_limit == source_length) {
-			str = String();
-			str.copy_from_unchecked(&get_data()[0], source_length);
+			str = *this;
```

Comment:
Definitely better. However, probably worth looking into the 0 allocation version, as instantiated a string is not needed here, we only need a start pointer and a size.

Reply:
Yeah, i was planning to address that once we have spans.